persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -382,4 +388,36 @@ private static Relation computeRelation(String fieldName, Entity entity, Entity
     private static String stripEscapeCharacter(String fieldName) {
         return fieldName.startsWith(BalSyntaxConstants.SINGLE_QUOTE) ? fieldName.substring(1) : fieldName;
     }
+
+    public static Path getSchemaFilePath(String sourcePath) {
+        List<Path> schemaFilePaths;
+
+        Path persistDir = Paths.get(sourcePath, PersistToolsConstants.PERSIST_DIRECTORY);
+        if (!Files.isDirectory(persistDir, LinkOption.NOFOLLOW_LINKS)) {
+            errStream.println("ERROR: the persist directory inside the Ballerina project does not exist. " +
+                    "run `bal persist init` to initiate the project before generation");
```

Comment:
Don't print any messages in this utils function. throw them as BalException and handle in the execute method and print it there.